app/controllers/cloud_network_controller.rb

```diff
@@ -190,49 +189,58 @@ def new
     @cloud_tenant_choices = {}
     CloudTenant.all.each { |tenant| @cloud_tenant_choices[tenant.name] = tenant.id }
 
-    drop_breadcrumb(
-      :name => _("Add New %{model}") % {:model => ui_lookup(:table => 'cloud_network')},
-      :url  => "/cloud_network/new"
-    )
+    drop_breadcrumb(:name => _("Add New Cloud Network"), :url => "/cloud_network/new")
   end
 
   def update
     assert_privileges("cloud_network_edit")
     @network = find_by_id_filtered(CloudNetwork, params[:id])
-
+    options = edit_form_params
     case params[:button]
     when "cancel"
-      cancel_action(_("Edit of %{model} \"%{name}\" was cancelled by the user") % {
-        :model => ui_lookup(:table => 'cloud_network'),
-        :name  => @network.name
-      })
+      cancel_action(_("Edit of Cloud Network \"%{name}\" was cancelled by the user") % { :name  => @network.name })
 
     when "save"
-      options = edit_form_params
-      begin
-        @network.update_network(options)
-        add_flash(_("Updating %{model} \"%{name}\"") % {
-          :model => ui_lookup(:table => 'cloud_network'),
-          :name  => @network.name
-        })
-      rescue => e
-        add_flash(_("Unable to update %{model} \"%{name}\": %{details}") % {
-          :model   => ui_lookup(:table => 'cloud_network'),
+      if @network.supports_update?
+        task_id = @network.update_cloud_network_queue(session[:userid], options)
+        add_flash(_("Cloud Network update failed: Task start failed: ID [%{id}]") %
+        {:id => task_id.inspect}, :error) unless task_id.kind_of?(Integer)
+        if @flash_array
+          javascript_flash(:spinner_off => true)
+        else
+          initiate_wait_for_task(:task_id => task_id, :action => "update_finished")
+        end
+      else
+        add_flash(_("Couldn't initiate update of Network \"%{name}\": %{details}") % {
           :name    => @network.name,
-          :details => e
+          :details => @network.unsupported_reason(:delete)
         }, :error)
       end
+    end
+  end
 
-      session[:edit] = nil
-      session[:flash_msgs] = @flash_array.dup if @flash_array
-      javascript_redirect previous_breadcrumb_url
+  def update_finished
+    task_id = session[:async][:params][:task_id]
+    cloud_network_id = session[:async][:params][:id]
```

Comment:
@gildub I'd prefer to leave it in, as it was removed upstream in a separate PR; I don't want to introduce potential future merge conflicts.  Let me know what you think!